misc.c

```diff
@@ -3063,3 +3063,42 @@ lib_contains_symbol(const char *path, const char *s)
 	return ret;
 #endif /* HAVE_NLIST_H */
 }
+
+ssize_t
+write_msg(const char *str, int flags)
+{
+	int output;
+	int output_fileno;
+	int needs_close;
+	ssize_t ret;
+
+	needs_close = 0;
+	output_fileno = (flags & WRITE_MSG_STDOUT) ? STDOUT_FILENO : STDERR_FILENO;
+
+	if (isatty(output_fileno)) {
+		output = output_fileno;
+	} else if ((output = open(_PATH_TTY, O_WRONLY)) != -1) {
+		needs_close = 1;
+	} else {
+		if (flags & WRITE_MSG_REQUIRE_TTY) {
+			errno = ENOTTY;
+			return -1;
+		}
+
+		output = output_fileno;
+	}
```

Comment:
I'm not convinced this section is necessary or correct. I guess it's mimicking the behaviour of `read_passphrase()`, but it isn't complete as that function will also call out to `$SSH_ASKPASS` and has some additional logic besides. 

You might be better off using the readpass.c `notify_start()` API, as it tries to do all this.

Reply:
This is done too now, not sure if it can be done better.